src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -55,7 +55,7 @@ public function loadUserByUsername($username)
                     throw new \InvalidArgumentException(sprintf('You must either make the "%s" entity Doctrine Repository ("%s") implement "Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface" or set the "property" option in the corresponding entity provider configuration.', $this->classOrAlias, get_class($repository)));
                 }
 
-                @trigger_error('Implementing loadUserByUsername from Symfony\Component\Security\Core\User\UserProviderInterface is deprecated since version 2.8 and will be removed in 3.0. Implement the Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface instead.', E_USER_DEPRECATED);
+                @trigger_error('Having a Doctrine repository implementing Symfony\Component\Security\Core\User\UserProviderInterface for loading users is deprecated since version 2.8 and will not work in 3.0. Make the repository implement Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface instead.', E_USER_DEPRECATED);
```

Comment:
I thought about `implementing in`, looks indeed better after reading it :).

About the part you removed, just implementing `UserProviderInterface` in a doctrine repository is still ok (no matter the why) so I think it makes clearer to mention that the deprecation is only about when using the entity provider. From what you given, what do you think about

> Implementing Symfony\Component\Security\Core\User\UserProviderInterface in a Doctrine repository for loading users through an entity provider is deprecated since version 2.8 and will not work in 3.0. Make the repository implement Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface instead.

?